### DISCUSSION TITLE
not a dxcore issue but a heads up on phantom DD availability

### DISCUSSION THREAD

**Discussion by ObviousInRetrospect** [SpenceKonde/DxCore#296] [General]
You might want to include any of this as a caveat to your 'DDs shipping' comments:

Don't trust microchipdirect's availability or ship dates on the avr32dd14 (and probably other DDs).

I had ordered 125 AVR32DD14-I/SL backk on 6/18, which they still say "Order now, up to 1,876 can ship on 09-Jul-2022". They sent me a confirmation saying they would ship on 6/25. on 6/27 they sent a delay notice and on 6/28 it said to contact them.

I discovered on my second call with them that they had placed this on a "planning hold" and don't know when they will be able to ship. I wish I knew this before I reved 3 designs to use this chip (based on the fact that I thought I could actually get it). PCBs on my bench, no eta on chips :-(. And due to their swapping of power and ground its not like I can substitute.

**Comment by SpenceKonde** [comment]
Tell me about it. (see attached image).
I'm in line a little bit ahead of you for some of the first low pincount dd's. It sounds like the first chips came off the production line are severely broken such that even their 3/4ths assed QA declared them unfit. After a debacle where unusable DA32's made it into customer hands. However, since that release, everything seems to be coming out slower, but the initial errata lists have a table of contents that fits on a single page (and the 162x tinies first shipped as a *REV E* - a Rev D almost shipped and was rejected after the pre-release errata was posted, and they only shipped the E.. So maybe someone's cracking the whip at the QA folks now). 

One might expect that at that point, since they're also calibrating the temperature sensor, that this would be where they binned the dies into industrial vs extended temp parts, but I find that hard to reconcile with the fact that there's no bit in sigrow nor marking on the package to tell you if you have an I-spec or E-spec on Dx-series (and the difference in temp range is the big one, 85C vs 125C. The E-spec also overclocks better at room temp, which isn't surprising). The tinies have the N or F after the part number indicating temp range, but the Dx parts don't, you have to ask microchip support about the week of mfg and lot code to find out, which is very strange IMO. 

In any event,  It's hard to see what the problem could be, since there is so little that is different on those parts vs the DD28/32, just less pins, Could they have had PC0 brought out to the pin instead of VDDIO2? (that should be easily fixed with no die change, just different bonding wire connections)? Is there a problem with the crystal oscillator driver, which has an added complication on the low pincount parts in that it can be either the 32k crystal or an HF crystal) on PA0/1? That's at least plausible.....  If it's a problem with the new PORTMUX options that's present on all DD's and was only noticed when that was the only mux option they had on the small ones, they need to be slapped upside the head, everyone on the design and test teams, because they've been here before. They have managed to ship with at least one portmux feature broken on every family of parts released since the new architecture came out, and you'd really think verifying that would be on their checklist, especially being as it's simple to do....  

If you consider how long it would take to get most other chips "order now, ships in July 2023"  and how much those dates slip (my backordered 412s ordered from mouser originally scheduled for September of 23 are now May of '24. lol...) (I'm annoyed with myself, I ordered last September thinking the est. Ship date was the end of that month, not the end of that month in the following year, I have other parts stranded in that order, and because I paid via PayPal they can't modify the order, all they can do is cancel the order and I'd lose my place in line, and I want the 25 3226's in qfn that came with it - they show as being in stock (I think they were set aside from the rationed lot for my order). They stopped answering my emails about it a while ago), an "Order now and it ships in a few weeks" sounds downright speedy :-P I am very annoyed over how microchip and distributors have handled the QFN 3226's.Or as the case is, how they *haven't* handled them except in trays of 490 at a time. Making them basically unobtainable to individuals. I will be lucky to have either item in that order by this time next year :-(

At least in the case of my boards, I have to Rev the DD20 board *anyway*, because I omitted the v-groove and it's on  a panel with 3 other designs (covered in photo by the partially assembled DD14 boards). 

So far, 1 of those designs is demonstrated defective (I forgot to connect two critical pins in an area of the board too crowded to execute a fix. One of them is the power for the USB physical layer, and the other is power for the PLL. Turns out USB serial adapters need those things powered. (it's a quad serial adapter via FT4232, breaking out all serial and modem control pins on all 4 ports with pullups selectable to 3.3 vs 5 V, a switch to flip PORT1 from serial port to UPDI on PORT0's CTS line (all my recent AVR boards have a CTS to set of solder bridge jumpers defaulting to UPDI, but with options for ground or the default USART0 XDIR, There's also a jumper to switch DTR from being autoreset to going to the USART0 XCK pin, with the idea of making it easier to do combined UPDI+serial and RS485 using the same header), and with jumpers that let you permanently turn port 0 and 3 into UPDI programmers). It wouldn't even say "the last device you connected has malfunctioned") and needs respin too. 
I'd be lying if I said I was optimistic about the other two (can't test one until i can make it up town to grab a few parts I forgot) though I got a brilliant idea for something to replace the other one on the panel that'll be way better - except to fit it i need some AVR32DD20 in the QFN 20, and just like before, their ordering policies for individuals are "fuck off, we don't sell to insects who can't afford to order 490". I'm mighty tempted to buy a 490 tray, wait until they're all hopelessly backordered, and go into the chip scalping biz. Shoulda done that for the 3226's, still kicking myself over that, I have a close to 100 bare boards manufactured that need 3226's, but it's looking like I can't get them before the end of the year if at all, and might not be able to get the in less than quantities of 490 even then. >.<

Do they think that only big companies can solder QFN? T-962/962A aren't that expensive, and while they struggle with the temperatures needed for 217C SAC305 (and just forget about the stuff with less silver), they work fine for leaded and lower melting solders.  With 158C ROHS solder (I have finally after trying close to 30 kinds of solder from a dozen brands, found a lead-free 158 one that seems very good, a "183" lead free that definitely melts below 183C, but makes beautiful joints, and have a "189 lead free" coming (after another brand of "189C lead free"solder than I got turned out to be silver bearing tin lead solder. Excellent tin lead solder, maybe the best I've used, but not lead free). So 1 good candidate and 2 that warrant further exploration, all with melting points above 150C. QFN goes on super easy with the good 158C stuff - much, much easier to rework when some inevitably come out crooked.

I now have a long document on solder that I'll be posting soon, discussing techniques, and solder + flux formulations and their pros and cons, and the scams and dishonesty I've encountered while buying it, and what my go-to fluxes are (I have one for hard tasks and another for less demanding ones that cleans easier) and what brands of solder are trash and what ones are good. 

![20220702_194114](https://user-images.githubusercontent.com/6779483/177220772-79317fa9-4d64-46d8-abc9-c6a48d85512f.jpg)

**Comment by ObviousInRetrospect** [comment]
And it looks like they shipped me the DD14's. If you don't get yours soon I'm happy to send you a few.

**Comment by obdevel** [comment]
Similar story with the DD20s. Ordered some, got the boards made and then the order is marked 'delayed'. Support tell me the order has shipped but I don't believe them as it still shows delayed. The part page shows AVR32DD20 available to ship on 19 July 22.

I've been waiting for this part to enable a project that needs a tiny board. Nothing else would be small enough.

Do I win the prize for the tiniest board ? ;) I've happily assembled 4x4mm parts before (ATSAMD11) so this shouldn't be too difficult. Looking forward to the soldering doc.

![board1](https://user-images.githubusercontent.com/13898399/177380006-93e6335b-ba7d-4c3d-8e81-2932af4659a7.png)

**Comment by SpenceKonde** [comment]
Is that the right size for the part outline? According to the datasheet
it's coming in the 3x3mm qfn like tinyavr, not the 4x4 one the product
brief suggested. 0.4mm pitch, not 0.5mm. The 28 pin qfn is a 4x4, and the
32 pin one is the same 5x5 0.5mm pitch package that the tiny 861 is made in
(using all the pins instead of just 5/8ths of them. Never understood why
they used 32 pin qfn for those)

Among other things that means you can make a 20-pin dip adapter pcb that
fits a dip20 socket

____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore: Arduino support for almost every ATTiny microcontroller
Contact: ***@***.***

**Reply by obdevel** [reply]
Yes, the DD20 is 3mm (REB package code, datasheet p598)). The SAMD11 is 4mm QFN28, so a comparable soldering challenge.

I have a programming board that attaches to the tiny project board via an edge connector and breaks out UPDI and USB/serial. And a generic breakout board so I can do some testing of all pins as well as the functionality that the project board doesn't implement.

**Comment by SpenceKonde** [comment]
IMHO the REBs are not nearly as hard to solder as they sound. as long as you're not using and iron (though modified footprints to facilitate hand soldering can help), at least. I was quite satisfied with the results of hot air soldering with Relife-GLON 158C paste. Supposedly lead free, but no composition listed but an ROHS logo. Same stuff did very well in reflow (and was markedly easier to rework  where bridges were found than tin-lead - that 15 degrees makes more of a difference than you'd think). And you can of course hotplate anything with a temp controlled hotplate; hot air and hotplate are nice since you can fix it as soon as you see a component snap into the wrong position as the solder paste under it melts. (bridges are IMHO easier cleared with an iron wielding either knife tip or tinned-face-only bevel tip. Knife tip is much faster and is suitable for most soldering tasks (it works miracles on smt resistors and capacitors - you hit both ends at once making it easy to replace, remove, or realign caps and resistors), tinned face only bevel is marginally better, but not as fast and not as generally useful, so I rarely bother with it.)  I've had trouble getting my reflow oven to pull off the temperatures needed for SAC305, one batch of SAC305 I got was total trash, with profoundly disappointing results, on the hotplate which normally works like a charm with SAC305,  and I gather this is a common problem with cheaper reflow ovens


 (I'm planning to melt a blob of that relife paste, corrode it (maybe galvanically, though I'd need to scare up a graphite rod, the goal being to generate enough soluble lead (if there's any lead in it) for a lead swab to get an unambiguous result from, but you'd rather not use an anode that would also corrode.  It's not the best analytic method, but I'm not spending a few hundred a pop to get the three supposedly lead free solders that are acceptable as solder tested (and I have at least twice as many that claimed to be lead free but aren't (one of which is possibly the best leaded solder I've used - leaded with 0.2% silver according to label, and it *looks* like T5 rather than T4 - but I'm trying to move to lead-free solder as much as possible), and ones that do appear to be lead free, but are crap solder, and even a couple of batches of leaded crap (how do you manage to make leaded solder that sucks that bad? I don't know.... but Conduction managed it, as well as the worst of the lead-free intermediate temperature solders. It's gotta be lead free, because it's such unspeakable garbage as solder that there's no way anyone would bother trying to sell it if it wasn't lead free (see, there's some demand for solder that melts at temperatures not higher than tin-lead solder, but well above the 138C of tin-bismuth eutectic solder., which is also only mediocre solder). After trying all these weird solder pastes, if you handed me an unmarked sample of solder paste and told me it was lead free, I would be highly skeptical of your claim if it turned out to also be good solder, whereas if it was crap solder, I'd figure you were probably telling the truth about it being lead free. I have twice had sellers swear that solder was lead free, then I get it and it says right on the package what alloy it is - Sn63Pb37, or Sn62.8,Pb37Ag0.2 - that's the very good leaded paste I mentioned above. 2uul snk 189C solder paste (though I don't know where they got that melting point, silvered tin lead solder should be 179?. Did a lovely job. I ran a board with no stencil by the "thin layer of paste in the general area of the pads, reflow, and then clean up the mess", and it came out a lot better than I was expecting. Of course, I'd neglected to run power to Vpll or Vphy of the FT4232, nor could I come up with a way to hack that on due to the density of the board there (I had to switch to an obnoxiously small crystal to make it fit for the next rev 😠 - I'd been hoping to be able to burn some of my 100 12 mhz larger than most designs call for crystals that way (the ones from ECS that are like that mini-HC/59), had to switch to an 3225 crystal  (which are luckily only about 50% more expensive from western suppliers (the china discount varies from 0% or worse for semiconductors not made by chiense firms, to as much as 95% (LEDs and switches/buttons are the places you generally see the biggest china discount, btw. Never buy a button from digikey, with notable anomalies like the presumably coutnerfeit MAX662's, which work great in spite of it - those cost like 50 cents vs the insane price maxim wants for the real ones. At that price, I mgiht as well just buy a dc-dc boost converter!)).... but I digress. .  But yeah, because I screwed up the design it's hard to say that it produced a working board after only minor cleanup, but I'm confident that if i'd powered the chip's clock generation and USB interface, it would have worked.)

**Comment by obdevel** [comment]
Yay! I have a TNT tracking number for my order, although it still shows as delayed. I wonder what will actually arrive and when.

I must admit to using a lot of Chinese stuff in my projects: jellybean components, solder, paste, flux, etc. I do most of my assembly on a hotplate although I have access to an oven at our local maker space, which I plan to use to experiment with double-sided assembly. That will mean differential solder temperatures, so your advice is very relevant. Any double-sided work I do now is really just a couple of components on the bottom, which I can hand solder.

I don't recall whether you plan to have a board for the DD32 but if not, I'm happy to help out with test coverage over and above what my current project will touch.

Here's the SAMD11 board I did a couple of years ago. Sadly, the Arduino core for this chip is no longer maintained (which may be problematical for Arduino as they use this chip (and core) as the USB/serial bridge on the Nano Every).

It's not a particularly practical chip as 40% of the 4K SRAM is consumed by the USB stack.

![IMG_0158](https://user-images.githubusercontent.com/13898399/177566738-3d5eaac7-08d1-4d62-b38c-4a4c8650b882.JPG)

**Comment by SpenceKonde** [comment]
Azduino nano dd and db will launch simultaneously.

Yeah, I use Chinese parts too. Anyone who buys a button from digikey is a
fool (same goes for someobe buying avrs on ebay especially in a shortage -
there are a lot of fake tinies, t85s and 412s mostly that are tiny13's that
have had the old name removed and rewritten, or are defective parts which I
think microchip attempted to get rid of after a manufacturing mishap.

____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore: Arduino support for almost every ATTiny microcontroller
Contact: ***@***.***

**Reply by obdevel** [reply]
Nobody sneeze !

![IMG_0160](https://user-images.githubusercontent.com/13898399/178034134-5095e2e4-a3bd-43bd-86b9-f71d61a83cd3.JPG)
